Src/Base/AMReX_VisMF.cpp

```diff
@@ -86,20 +95,81 @@ VisMF::Initialize ()
     pp.query("usedynamicsetselection", useDynamicSetSelection);
     pp.query("iobuffersize", ioBufferSize);
     pp.query("allowsparsewrites", allowSparseWrites);
+    pp.query("noutfiles", newOutFiles);
+    pp.query("asyncwrites", nAsyncWrites);
+
+    async_comm.assign(nAsyncWrites, MPI_COMM_NULL);
+
+    VisMF::SetNOutFiles(newOutFiles);
 
     initialized = true;
 }
 
 void
 VisMF::Finalize ()
 {
+
+#ifdef AMREX_MPI_MULTIPLE
+    VisMF::asyncWaitAll();
+
+    for (int i=0; i<nAsyncWrites; ++i)
+    {
+        if (async_comm[i] != MPI_COMM_NULL)
+        {
+            MPI_Comm_free(&async_comm[i]);
+        }
+    }
+#endif
+
     initialized = false;
 }
 
 void
-VisMF::SetNOutFiles (int noutfiles, MPI_Comm comm)
+VisMF::SetNOutFiles (int newoutfiles, MPI_Comm comm)
 {
-    nOutFiles = std::max(1, std::min(ParallelDescriptor::NProcs(comm), noutfiles));
+
+    const int nranks = ParallelDescriptor::NProcs(comm);
+    const int myproc = ParallelDescriptor::MyProc(comm);
+
+    // Must be called globally with this change (MPI_Comm_split on m_comm == MPI_COMM_WORLD)
+    // So, minimize when it's done?? How?? (Comm / newoutfiles could both change!)
+    nOutFiles = std::max(1, std::min(nranks, newoutfiles));
+
+#ifdef AMREX_MPI_MULTIPLE
+
+    VisMF::asyncWaitAll();
+    current_comm = 0;
+
+    // Function is being recalled, so free MPI objects. 
+    if (initialized)
+    {
+        for (int i=0; i<nAsyncWrites; ++i)
```

Comment:
If the user calls SetNOutFiles with the same `noutfiles`, are there any reasons to free the comms and recreate them?